pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/AccessorMethodGenerationRule.java

```diff
@@ -39,41 +38,35 @@ public void end(RuleContext ctx) {
 
     @Override
     public Object visit(ASTFieldAccess node, Object data) {
-        JFieldSymbol sym = node.getReferencedSym();
-        if (sym != null && sym.getConstValue() == null) {
-            checkMemberAccess((RuleContext) data, node, sym);
-        }
+        checkMemberAccessIfConstValueIsNull(node, (RuleContext) data, node.getReferencedSym());
         return null;
     }
 
     @Override
     public Object visit(ASTVariableAccess node, Object data) {
-        JVariableSymbol sym = node.getReferencedSym();
-        if (sym instanceof JFieldSymbol) {
-            JFieldSymbol fieldSym = (JFieldSymbol) sym;
-            if (((JFieldSymbol) sym).getConstValue() == null) {
-                checkMemberAccess((RuleContext) data, node, fieldSym);
-            }
-        }
+        checkMemberAccessIfConstValueIsNull(node, (RuleContext) data, node.getReferencedSym());
         return null;
     }
 
+    private void checkMemberAccessIfConstValueIsNull(JavaNode node, RuleContext data, JVariableSymbol sym) {
+        if (sym instanceof JFieldSymbol && ((JFieldSymbol) sym).getConstValue() == null) {
+            checkMemberAccess(data, (ASTExpression) node, (JFieldSymbol) sym);
+        }
+    }
+
     @Override
     public Object visit(ASTMethodCall node, Object data) {
-        JMethodSymbol symbol = (JMethodSymbol) node.getMethodType().getSymbol();
-        checkMemberAccess((RuleContext) data, node, symbol);
+        checkMemberAccess((RuleContext) data, node, node.getMethodType().getSymbol());
         return null;
     }
 
     private void checkMemberAccess(RuleContext data, ASTExpression node, JAccessibleElementSymbol symbol) {
-        checkMemberAccess(data, node, symbol, this.reportedNodes);
+        checkMemberAccess(data, node, symbol, reportedNodes);
     }
 
     static void checkMemberAccess(RuleContext ruleContext, JavaNode refExpr, JAccessibleElementSymbol sym, Set<JavaNode> reportedNodes) {
-        if (Modifier.isPrivate(sym.getModifiers())
-            && !Objects.equals(sym.getEnclosingClass(),
```

Comment:
no need to wrap as we have `max_line_length = 140` in place. https://github.com/pmd/pmd/pull/5635/files